getAndWatch_stream_provider.go

```diff
@@ -106,13 +106,16 @@ func (p *GetAndWatchStreamProvider) sendLoop(strm grpc.ServerStream, peer Peer)
 	defer p.metrics.clientCounter.Dec()
 	broadcaster := p.broadcaster
 	streamCh := make(chan *mux.StateUpdate, p.config.SubscriberInputBufferLen)
-	broadcaster.Register(streamCh, func(config *mux.ConsumerConfig) error {
+	err := broadcaster.Register(streamCh, func(config *mux.ConsumerConfig) error {
 		config.OnBackpressure(func(interface{}) {
 			p.config.OnBackPressure(streamName)
 			p.metrics.backPressureCounter.Inc()
 		})
 		return nil
 	})
+	if err != nil {
+		Log.Fatal("failed to register to broadcaster", zap.Error(err))
```

Comment:
Is it correct to os.Exit() here?
sendLoop() is used by stream_registry.go/publishOnStream(), which can return an error.
I think an error should be propagated to the caller, rather than stopping the application.

Reply:
The API is super strange, but here there should never be any error. So I think we can do a fatal here for now